pkg/diagnose/firewall.go

```diff
@@ -104,46 +105,21 @@ func spawnSnifferPodOnNode(client kubernetes.Interface, nodeName, namespace, pod
 	return spawnPod(client, scheduling, "validate-sniffer", namespace, podCommand, imageRepInfo)
 }
 
-func getActiveGatewayNodeName(clusterInfo *cluster.Info, hostname string, imageRepInfo *image.RepositoryInfo,
-	status reporter.Interface,
-) (string, error) {
-	nodes, err := clusterInfo.ClientProducer.ForKubernetes().CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{
-		LabelSelector: "submariner.io/gateway=true",
-	})
+func getActiveGatewayNodeName(clusterInfo *cluster.Info, status reporter.Interface) (string, error) {
```

Comment:
This is a good optimisation @tpantelis. This would fail if the `submariner-gateway` pod is not running, but hey that's what diagnose is all for :)